internal/provider/resource_gitlab_project_test.go

```diff
@@ -883,6 +891,111 @@ func TestAccGitlabProject_CreateProjectInUserNamespace(t *testing.T) {
 	})
 }
 
+func TestAccGitlabProject_InstanceBranchProtectionDisabled(t *testing.T) {
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabProjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					settings, _, err := testGitlabClient.Settings.GetSettings()
+					if err != nil {
+						t.Fatalf("failed to get settings: %v", err)
+					}
+					t.Cleanup(func() {
+						if _, _, err := testGitlabClient.Settings.UpdateSettings(&gitlab.UpdateSettingsOptions{DefaultBranchProtection: gitlab.Int(settings.DefaultBranchProtection)}); err != nil {
+							t.Fatalf("failed to update instance-wide default branch protection setting to default: %v", err)
+						}
+
+						// NOTE: without waiting a few seconds, the instance-wide default branch protection setting is not properly taken into account
+						//       and the test will fail. I'm not sure what's going on here on GitLab side, but it seems to be another example of
+						//       something that is happening in the background when calling the API.
+						//       See upstream issue: https://gitlab.com/gitlab-org/gitlab/-/issues/364812
+						time.Sleep(time.Second * 60) // lintignore: R018 // see comment above
```

Comment:
I've been using the following naive loop to test stability of this code - in case you want to play around a little ;) :

```sh
for i in $(seq 50); do make testacc RUN="\(TestAccGitlabProject_InstanceBranchProtectionDisabled\|TestAccGitlabProject_ImportURLMirrored\)"; [[ $? -ne 0 ]] && echo "Broken after $i attempts" && break || echo "Attempt $i ok"; done
```

I've also set `-count=1` for the `go test` cmd in the makefile ...

Reply:
Yikes, this is brutal.  We can't even add a "wait for" loop I assume because even though the setting isn't applied it likely comes back from the API as disabled at this point.

I remembered seeing a hidden environment variable when we were looking at the runner token reset issue (which required a similar janky wait until we figured out we could set it on startup, which wouldn't work here). It looks like there is a variable you can set to disable in memory application settings which may be worth a try here? https://gitlab.com/gitlab-org/gitlab/-/blob/master/spec/requests/api/ci/runners_reset_registration_token_spec.rb#L81

Reply:
@timofurrer  - I did managed to get GitLab to start up, but that was before we added the additional ruby work related to application_settings, which seems like it's where it's erroring. I wonder if it works if some of those are removed, just for experimentation sake. I can probably test some of that this weekend too.

Reply:
Hey @timofurrer  - I tested that `IN_MEMORY_APPLICATION_SETTINGS` as `true`, and I see the same error that you do, but when I swap it to `false` (which is what I think we want so it doesn't try and "cache" settings in memory) it boots up appropriately

![image](https://user-images.githubusercontent.com/50960557/173200251-aabdd28a-88fc-4472-8fa7-7f70c868ce68.png)

![image](https://user-images.githubusercontent.com/50960557/173200261-d9bb1d9e-6056-485d-9abf-b57530aeeb5b.png)

When I use `false`, I also see passing on the very first test run:

![image](https://user-images.githubusercontent.com/50960557/173200276-840bebe1-ada3-404b-be63-7552b32f9d2e.png)

Reply:
Can you keep it running for a while? Maybe also producing some load. 
I cannot reproduce it. The only difference I spot is that I'm using an EE instance for my tests ... 🤔 

Also, without setting the `IN_MEMORY_APPLICATION_SETTINGS` - it may pass for me quite a bunch of times in a row, but then eventually fails :'(

Reply:
@PatrickRice-KSC and just to make sure, you've removed the sleeps, right? ... just happened to me before that I didn't 🤦

Reply:
I've now used `gitlab_rails['application_settings_cache_seconds'] = 0` to disable the caching as suggest [here](https://gitlab.com/gitlab-org/gitlab/-/issues/364812#note_986366898).

Let's keep 🤞 that all tests still pass!

Reply:
> You are running a CE instance, for which the ImportURLMirrored tests are not executed ;)

🤦 - good call. hopefully that workaround works!

Reply:
@PatrickRice-KSC It actually looks good :)

Do you mind giving it another review?

Reply:
Stills seems to fail ... but more rarely I would say: https://github.com/gitlabhq/terraform-provider-gitlab/runs/6845475259?check_suite_focus=true#step:8:243

Reply:
@timofurrer  - I suspect this is failing periodically in CI because it's relying on the updates in `gitlab.rb`, and those are used from the main branch instead of the PR branch. Want to create a PR to merge those changes into the main branch, and I can approve that then we can rebase and try again?

Reply:
@PatrickRice-KSC oh boy ... absolutely. Yes, I'll do that!

Reply:
@PatrickRice-KSC Please approve and merge: https://github.com/gitlabhq/terraform-provider-gitlab/pull/1130

Reply:
@PatrickRice-KSC looking good now! I think we can risk to merge this. WDYT?

Reply:
@timofurrer  - Agreed, I'll get this merged 😄